src/previous.js

```diff
@@ -0,0 +1,40 @@
+import { parse, stringify } from 'qs';
+
+/**
+ * This function navigates to a location stored in the 'previous' query param or the
+ * "All institutions" asset list, if there is none.
+ *
+ * @param push history.push()
+ * @param search location.search
+ */
+export const navigate = ({ push }, { search }) => {
+  const { previous } = parse(search, { ignoreQueryPrefix: true });
+  push(previous ? previous : '/assets');
+};
+
+/**
+ * Uses the DOM to parse the search part of a url.
+ */
+const get_search = (url) => {
+    const a = document.createElement('a');
+    a.href = url;
+    return a.search;
+};
+
+/**
+ * If there is a department, then this function adds a 'previous' query param to the given url
+ * of the department's assert list url.
+ *
+ * @param url url to modify
+ * @param department INSTID of department or null.
+ * @returns {*} modified url
+ */
+export const encode_search = (url, department) => {
```

Comment:
This can be replaced with things such as ``URLSearchParams.append()`` and ``URLSearchParams.toString()`` I think.